libraries/ESP8266WiFi/src/ESP8266WiFiGeneric.h

```diff
@@ -77,6 +77,26 @@ class ESP8266WiFiGenericClass {
         uint8_t channel(void);
 
         bool setSleepMode(WiFiSleepType_t type, uint8_t listenInterval = 0);
+        // ESP32 compatibility
+        bool setSleep(bool enable)
+        {
+            if (enable)
+            {
+                return setSleepMode(WIFI_LIGHT_SLEEP);
+            }
+            else
+            {
+                return setSleepMode(WIFI_NONE_SLEEP);
+            }
+        }
+        bool setSleep(wifi_ps_type_t mode)
+        {
+            return setSleepMode((WiFiSleepType_t)mode);
+        }
+        bool getSleep()
+        {
+            return getSleepMode() == WIFI_LIGHT_SLEEP;

```

Comment:
`getSleepMode() != WIFI_NONE_SLEEP`
(so LIGHT and MODEM are treated as sleep mode enabled)

Reply:
I thought about that, but see my comment above. To maintain ESP32 compatibility I did it the same way that it's on ESP32:
https://github.com/espressif/arduino-esp32/blob/3fe7c2e8cdcc34af0f5d3c55f8b4cb79ef5318cf/libraries/WiFi/src/WiFiGeneric.cpp#L669

Reply:
One other option would be to drop the bool version

Reply:
Indeed, the comment there is:
```cpp
// @return true if modem sleep is enabled
bool WiFiGenericClass::getSleep()
```
The naming is confusing, because "light sleep" is also an enabled sleep mode while "none sleep" has sleep mode disabled.
What about replicating the comment in esp8266 .h file, insisting on that it reflects the esp32 API ?

> One other option would be to drop the bool version

That would defeat the goal of this PR. We just don't need to add it in esp8266 documentation.
esp32 libraries will compile, esp8266 .h readers will see the comment, esp8266 api is unchanged.